### ISSUE TITLE
BUG: Data store error - AWS batch/step execution

### ISSUE THREAD

**Issue by itLovaz** [Netflix/metaflow#2137]
**Environment:**
metaflow version: 2.12.29
Python 3.11 (Docker Image from public.ecr.aws/docker/library/python:3.11)
Running on AWS Batch

**Description:**
Tested with version 2.12.28 and it runs successfully, with this latest version we get:
Data store error: No completed attempts of the task was found for task `MyFlow/sfn-*/_parameters/*-params`.

Maybe worth mentioning that we include a json file into `MyFlow` like:
```

json_config = IncludeFile(
        name="my_config",
        required=True,
        help="The Configuration",
        default=f"./{PARAMS_JSON}",
    )

```

**Comment by colebaileygit** [comment]
I also got this error when running on argo workflows. My flow does not use `IncludeFile` but just usual parameters

**Comment by gregdes** [comment]
And another confirmation with step on batch.  2.12.29 displays the error, 2.12.28 does not.

**Comment by gabriel-rp** [comment]
I also got this error on Argo Workflows. Same problematic version (`2.12.29`) and the same fix (downgrade to `2.12.28`).
